src/components/MenuItem.tsx

```diff
@@ -342,6 +352,25 @@ function MenuItem(
         isDeleted ? styles.offlineFeedback.deleted : {},
     ]);
 
+    useEffect(() => {
+        if (!shouldFocusTemporarily) {
+            return;
+        }
+        setTemporaryilyFocused(true);
+        tempFocusTimeoutRef.current = setTimeout(() => setTemporaryilyFocused(false), temporaryFocusDuration);
+    }, [shouldFocusTemporarily, temporaryFocusDuration]);
+
+    useEffect(
+        () => () => {
+            if (!tempFocusTimeoutRef.current) {
+                return;
+            }
+
+            clearTimeout(tempFocusTimeoutRef.current);
+        },
+        [],
+    );
```

Comment:
Can we extract this into a custom hook, since we'll have to use it in other places, ie BaseListItems when displaying added Categories, Tags, etc

Reply:
I can, but right now, I'm stuck on animating the icon within the menu item. Here's what I've done till now:

https://github.com/Expensify/App/assets/30054992/2ab65879-7a7c-4f7f-b736-5d29cea4f11d

Reply:
> but right now, I'm stuck on animating the icon within the menu item. Here's what I've done till now:

> * we also make the icon of the highlighted nav item use our success/green400 color

Unless I am missing something, isn’t making the following change [here](https://github.com/allroundexperts/Expensify/blob/bb2cce2ff2a67c58f851eb84977028cde01fc8e0/src/components/MenuItem.tsx#L499) good enough to achieve the above?

	getButtonState(temporaryilyFocused || focused || isHovered,

Here is a video:

https://github.com/Expensify/App/assets/3004659/bdb7e346-4baf-4937-b390-f51b01d49054